79 update config README

crispy-wonton opened the pull request:
Fixes #79 

# Description

Add new data sources and attributions to config/README

# Instructions for Reviewer

During your review, along with general review of readability /clarity etc, please could you check:
- all links work and take you to the expected dataset
- descriptions make sense
- you agree with removing the `S3 file` column. I removed it because I don't think it added value as all the S3 paths can already be found in the `config/base.yaml` and also because they are so changeable it would create unnecessary work to maintain the README.
- all the appropriate data attribution statements are there
- i haven't missed any datasets that should have been added

# Checklist:

- [ ] I have refactored my code out from `notebooks/`
- [ ] I have checked the code runs
- [ ] I have tested the code
- [x] I have run `pre-commit` and addressed any issues not automatically fixed
- [x] I have merged any new changes from `dev`
- [x] I have documented the code
  - [ ] Major functions have docstrings
  - [x] Appropriate information has been added to `README`s
- [x] I have explained this PR above
- [x] I have requested a code review

crispy-wonton committed:
update config/README with Scotland data
- add sources for feature and reweighting datasets
- add required data attribution statements
- remove 'S3 file' column for brevity and easier maintenance - the path is too changeable and it is already recorded in the base config file
- add new 'feature' column to feature datasets table
- update config key for NRS_households (from NRS_dwellings) to reflect actual dataset- we use the household not dwelling counts

crispy-wonton committed:
move footnotes in config/README for ease of reading

caldwellst commented on asf_heat_pump_suitability/config/README.md:
```diff
@@ -4,43 +4,51 @@
+| Listed building status        | `["W_cadw_listed_buildings"]`                              | [Cadw](https://datamap.gov.wales/layers/inspire-wg:Cadw_ListedBuildings)<sup>9</sup>                                                                                                                            | May 2024                          | Point geometries of listed buildings in Wales.             ...
+| Listed building status        | `["S_scottish_gov_listed_buildings"]`                      | [Scottish Government](https://www.data.gov.uk/dataset/722b93f3-75fd-47ce-9f06-0efcfa010ecf/listed-buildings)<sup>4</sup>                                                                                        | June 2024                         | Point geometries of listed buildings in Scotland.          ...
+| In building conservation area | `["E_historic_england_conservation_areas"]`                | [Historic England](https://www.planning.data.gov.uk/dataset/conservation-area)<sup>10</sup>                                                                                                                     | June 2024                         | Building conservation area boundaries in England (places o ...
+| In building conservation area | `["W_welsh_gov_conservation_areas"]`                       | [Welsh Government](https://datamap.gov.wales/layers/inspire-wg:conservation_areas)<sup>4</sup>                                                                                                                  | Sept 2022                         | Building conservation area boundaries in Wales (places of  ...
```
Beyond the scope of this PR, but just to note that the conservation areas you linked have now been flagged as `(old)` and there's a new dataset from November 28 2024 [here](https://datamap.gov.wales/layers/geonode:conservation_areas_wales)!

crispy-wonton replied:
Great flag, thank you!

crispy-wonton replied:
Have added to sprint backlog.

caldwellst commented on asf_heat_pump_suitability/config/README.md:
```diff
@@ -4,43 +4,51 @@
-| `["data_source"]["E_WPD_west_midlands_bounds"]` | `source_data/grid_operators/WPD/west-midlands-primary.gpkg` | [WPD West Midlands Network Areas](https://connecteddata.nationalgrid.co.uk/dataset/spatial-datasets) | Jun 2024 | Primary substation service area boundary polygons for West Midlands region. |
-| `["data_source"]["EW_lsoa_bounds"]` | `source_data/Lower_layer_Super_Output_Areas_2021_EW_BFE_V9_-9107090204806789093` | [ONS](https://geoportal.statistics.gov.uk/datasets/ons::output-areas-december-2021-boundaries-ew-bfe-v9/about)<sup>3</sup> | December 2021 | Lower Level Super Output Area boundaries EW BFE. |
+This table shows the different datasets used to calculate each feature and their sources. The 'Config key' column indicates the
+key used in `asf_heat_pump_suitability/config/base.yaml` under the primary `["data_source"]` key.
```
General note possibly outside this PR, but may want to consider instead of `Date` using `Date of access`? I assume most of these assets are static on S3 and not updated every run from source? Just a slightly more standard convention that also can be easier to maintain since you only update the date when you update your codebase/database. Not critical though.

crispy-wonton replied:
I think it's a good idea for us to include date of access. Will do this in a separate PR.

I think we should also keep the date of the dataset itself so that we can easily see how old the data is that we're using.

caldwellst commented on asf_heat_pump_suitability/config/README.md:
```diff
@@ -4,43 +4,51 @@
+| In building conservation area | `["W_welsh_gov_conservation_areas"]`                       | [Welsh Government](https://datamap.gov.wales/layers/inspire-wg:conservation_areas)<sup>4</sup>                                                                                                                  | Sept 2022                         | Building conservation area boundaries in Wales (places of  ...
+| In building conservation area | `["EW_ons_lsoa_lad_lookup"]`                               | [ONS](https://geoportal.statistics.gov.uk/datasets/ons::lsoa-2021-to-local-authority-districts-april-2023-best-fit-lookup-in-ew/explore)<sup>3</sup>                                                            | April 2023                        | LSOA to LAD lookup.                                        ...
+| In World Heritage Site        | `["S_historic_environment_scotland_world_heritage_sites"]` | [Scottish Government](https://www.data.gov.uk/dataset/eab6ee72-23e8-46df-b74b-c2a9cb3ee6e0/world-heritage-sites)<sup>4</sup>                                                                                    | June 2024                         | Geospatial polygons of UNESCO World Heritage Sites in Scot ...
+| Grid capacity                 | `["UK_poi_locations"]`                                     | [CDRC](https://data.cdrc.ac.uk/dataset/point-interest-data-united-kingdom)<sup>3</sup>                                                                                                                          | Sept 2024                         | Points of interest geographic locations and classification ...
```
lol "Site under maintenance". Will check back, but assume has nothing to do with the underlying data. Leaving this flag for myself.

crispy-wonton replied:
Is this the POI dataset? If so, seems to be working now: https://data.cdrc.ac.uk/dataset/point-interest-data-united-kingdom

caldwellst replied:
Yep, all good now!

caldwellst commented on asf_heat_pump_suitability/config/README.md:
```diff
@@ -4,43 +4,51 @@
+| Grid capacity                 | `["S_SSEN_sepd_bounds"]`                                   | [SEPD Network Areas](https://data.ssen.co.uk/@ssen-distribution/primary-substation-boundaries)                                                                                                                  | Nov 2023                          | Primary substation service area boundary polygons for SEPD ...
+| Grid capacity                 | `["E_SSEN_shepd_bounds"]`                                  | [SHEPD Network Areas](https://data.ssen.co.uk/@ssen-distribution/primary-substation-boundaries)                                                                                                                 | Nov 2023                          | Primary substation service area boundary polygons for SHEP ...
+| Grid capacity                 | `["E_UKPN_primaries"]`                                     | [UKPN Network Capacity Map](https://ukpowernetworks.opendatasoft.com/explore/dataset/ukpn_primary_postcode_area/table/)                                                                                         | Oct 2024                          | Primary substation data and service area mappings for Sout ...
+| Grid capacity                 | `["EW_WPD_capacity"]`                                      | [WPD Network Capacity Map](https://connecteddata.nationalgrid.co.uk/dataset/nged-network-capacity/resource/d1895bd3-d9d2-4886-a0a3-b7eadd9ab6c2)                                                                | 2024                              | Network capacity data for primary substations across WPD's ...
```
Hmm, 403 error forbidden. So might be valid, just I don't have access!

caldwellst replied:
Probably why the links below are to the general "Spatial datasets" page.

caldwellst commented on asf_heat_pump_suitability/config/README.md:
```diff
@@ -50,6 +58,19 @@ a. Rural-urban classification codes are mapped to text descriptions according to
+| `["S_census_accommodation_type"]`  | [Scotland's Census](https://www.scotlandscensus.gov.uk/search-the-census#/search-by)<sup>4</sup> - [National Records of Scotland](https://www.nrscotland.gov.uk/)<sup>13</sup> | Census 2022                   | Census 2022 household counts by property type per Data Zone in Scotland.                                                               |
+|       `["EW_census_tenure"]`       | [ONS](https://www.ons.gov.uk/datasets/TS054/editions/2021/versions/4)<sup>4</sup>                                                                                              | Census 2021, updated Mar 2023 | Census 2021 household counts by tenure per LSOA in England and Wales.                                                                  |
+|       `["S_census_tenure"]`        | [Scotland's Census](https://www.scotlandscensus.gov.uk/search-the-census#/search-by)<sup>4</sup> - [National Records of Scotland](https://www.nrscotland.gov.uk/)<sup>13</sup> | Census 2022                   | Census 2022 household counts by tenure per Data Zone in Scotland.                                                                      |
+|     `["EW_cdrc_dwelling_age"]`     | [CDRC](https://data.cdrc.ac.uk/dataset/dwelling-ages-and-prices/resource/dwelling-age-band-counts-lsoa-2015)<sup>11</sup>                                                      | 2015                          | Residential dwelling ages, grouped into approximately 10-year age bands from pre-1900 to 2015, with counts of each age group per LSOA. |
```
Hmm, dataset doesn't seem to load, says "The file could not be created" and searching for "dwelling" doesn't return anything immediately.

crispy-wonton replied:
Thanks! looks like they updated the path slightly - I've updated it here.

caldwellst commented on asf_heat_pump_suitability/config/README.md:
```diff
@@ -4,43 +4,51 @@
-| `["data_source"]["E_WPD_south_west_bounds"]` | `source_data/grid_operators/WPD/south-west-primary.gpkg` | [WPD South West Network Areas](https://connecteddata.nationalgrid.co.uk/dataset/spatial-datasets) | Jun 2024 | Primary substation service area boundary polygons for South West England region. |
-| `["data_source"]["E_WPD_west_midlands_bounds"]` | `source_data/grid_operators/WPD/west-midlands-primary.gpkg` | [WPD West Midlands Network Areas](https://connecteddata.nationalgrid.co.uk/dataset/spatial-datasets) | Jun 2024 | Primary substation service area boundary polygons for West Midlands region. |
-| `["data_source"]["EW_lsoa_bounds"]` | `source_data/Lower_layer_Super_Output_Areas_2021_EW_BFE_V9_-9107090204806789093` | [ONS](https://geoportal.statistics.gov.uk/datasets/ons::output-areas-december-2021-boundaries-ew-bfe-v9/about)<sup>3</sup> | December 2021 | Lower Level Super Output Area boundaries EW BFE. |
+This table shows the different datasets used to calculate each feature and their sources. The 'Config key' column indicates the
```
Agree with removing all the S3 pathing here, but maybe some more information here (or elsewhere in the repo?) would be helpful.

Where's this data stored? How is it updated? Are they static files updated only when we select, or are they updated in every pipeline run? I see where this doc is linked on the main README, but no real mention there, and some mention of EPC data coming from asf-daps in the pipeline README, but unclear here.

crispy-wonton replied:
Good points, thank you. I'll add this information.

caldwellst reviewed (changes requested):
My first review! Overall makes sense, and looks much cleaner than before!

However, might want to add a bit more detail on the storage and updating of the data, as that wasn't clear to me from reviewing this or searching through the repo initially (admittedly I didn't search for too long!). There's a few links I've highlighted that are dead or possibly incorrect.

Other than that, the one I did notice, which would likely just need y'alls discussion outside this PR, is a dataset that seems to have updated end of November that you may want to update in the pipeline.

crispy-wonton committed:
add information to config/README about data sources, EPC data, and fix broken links

crispy-wonton committed:
replace broken NRS dwellings source URL in `config/README.md`

crispy-wonton committed:
Merge branch 'dev' into 79_update_readme

crispy-wonton committed:
add new SSEN data sources for grid capacity in Scotland to config/README
- add attribution statements for grid capacity datasets

crispy-wonton committed:
replace dead link to ROS INSPIRE download service

crispy-wonton committed:
add licensing information for EPC to config/README

crispy-wonton committed:
add that EPC data has been deduplicated to config/README

crispy-wonton commented:
@caldwellst I've addressed/responded to your comments. Since your last review I have:

1. Added info on how the datasets are stored and update
2. Added info on where the EPC data comes from and licensing
3. Added missing data attribution/licensing statements for grid capacity datasets
4. Added 2 new grid capacity datasets: 'S_SSEN_demand' and 'SHET_SSEN_demand' config keys
5. Replaced broken/forbidden links 
6. I have checked all the links work

Please let me know if you are happy with the changes, especially 1 and 2.

caldwellst reviewed (dismissed):
All looks good now! Over to @lizgzil to just check if there are any outstanding datasets or clear issues with documentation that she would notice.

lizgzil commented on asf_heat_pump_suitability/config/README.md:
```diff
@@ -2,45 +2,68 @@
+
+| Feature                       | Config key                                                 | Source                                                                                                                                                                                                                                                                     | Date                              | ...
+| :---------------------------- | :--------------------------------------------------------- | :------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | :-------------------------------- | ...
+| LSOA, MSOA, LAD               | `["UK_ons_postcode_dir"]`                                  | [ONS](https://geoportal.statistics.gov.uk/datasets/487a5ba62c8b4da08f01eb3c08e304f6/about)<sup>1,2,3</sup>                                                                                                                                                                 | August 2023                       | ...
```
```suggestion
| LSOA, MSOA, LAD, RUC               | `["UK_ons_postcode_dir"]`                                  | [ONS](https://geoportal.statistics.gov.uk/datasets/487a5ba62c8b4da08f01eb3c08e304f6/about)<sup>1,2,3</sup>                                                                                                                                                                 | August 2023                       | ONS postcode directory for UK as at August 2023. Source of output area information and rural-urban classification<sup>a</sup>. |
```

lizgzil reviewed (dismissed):
@crispy-wonton I've read through the descriptions, checked against the config, and this looks great to me and makes sense to remove the s3 path. I'm not aware that you've missed anything. Thanks for being so thorough! 🎉

crispy-wonton committed:
add RUC to feature column in config/README
